src/sentry/integrations/gitlab/integration.py

```diff
@@ -242,9 +241,11 @@ def get_group_info(self, access_token, installation_data):
             access_token,
             installation_data['verify_ssl']
         )
-        resp = client.get_group(installation_data['group'])
-
-        return resp.json
+        try:
+            resp = client.get_group(installation_data['group'])
+            return resp.json
+        except ApiError:
```

Comment:
So, when I look at this, do we only want `"The requested GitLab group could not be found."` to show up when it's a `404` or any `ApiError`? I'm not entirely sure what conditions an `ApiError` is triggered, but I'm fairly certain it's not just `404`.

Reply:
There are other conditions that could cause an ApiError. Any connection issues or 500 error from GitLab would also become `ApiError`. We could check the status code and give different messages that way.

Reply:
Yeah I think we lose what the actual error is here. If you'd like to just present a particular message to the user that's ok, but we have to at least log the error. Since you're handling the error and throwing a different error, if it was a 500, it wouldn't show up in sentry as such.

Reply:
Oh I see nvm. Looks like this is just an apierror caused by gitlab. You meant 500 from GitLab. I read that wrong. I'm just very annoyed with the way errors are handled in integrations. They obscure what actually happened and you get "Unknown error" throwing away the actual error. Anyway good to know this isn't really the case here.